src/components/license-activation/LicenseActivation.jsx

```diff
@@ -1,27 +1,38 @@
-import React, { useContext } from 'react';
-import { Redirect, useParams, useLocation } from 'react-router-dom';
+import React, { useContext, useEffect, useState } from 'react';
+import { Redirect, useLocation } from 'react-router-dom';
 import { Helmet } from 'react-helmet';
 import { AppContext } from '@edx/frontend-platform/react';
 import { Alert, Container } from '@edx/paragon';
 
 import { LoadingSpinner } from '../loading-spinner';
-import { useLicenseActivation } from './data/hooks';
 
 import { useRenderContactHelpText } from '../../utils/hooks';
+import LicenseActivationErrorAlert from './LicenseActivationErrorAlert';
+import { UserSubsidyContext } from '../enterprise-user-subsidy/UserSubsidy';
 
 export const LOADING_MESSAGE = 'Your enterprise license is being activated! You will be automatically redirected to your organization\'s learner portal shortly.';
 
-export default function LicenseActivation() {
-  const { activationKey } = useParams();
+const LicenseActivation = () => {
   const { enterpriseConfig } = useContext(AppContext);
   const renderContactHelpText = useRenderContactHelpText(enterpriseConfig);
   const location = useLocation();
+  const fromLocation = location.state?.from;
+  const { activateUserLicense } = useContext(UserSubsidyContext);
+  const [activationSuccess, setActivationSuccess] = useState();
 
-  const [activationSuccess, activationError] = useLicenseActivation({
-    activationKey,
-    enterpriseUUID: enterpriseConfig.uuid,
-    autoActivated: !!location.state?.from,
-  });
+  useEffect(() => {
+    const activateLicense = async () => {
+      const autoActivated = !!location.state?.from;
```

Comment:
[question] Should this be `fromLocation`?

Reply:
nodders